sound/soc/soc-pcm.c

```diff
@@ -1203,6 +1203,15 @@ static int dpcm_be_connect(struct snd_soc_pcm_runtime *fe,
 	fe_substream = snd_soc_dpcm_get_substream(fe, stream);
 	be_substream = snd_soc_dpcm_get_substream(be, stream);
 
+	if (!fe_substream) {
+		dev_err(fe->dev, "%s: fe_substream not initialized for direction %d\n", __func__, stream);
+		return -EINVAL;
+	}
+	if (!be_substream) {
+		dev_err(be->dev, "%s: be_substream not initialized for direction %d\n", __func__, stream);
+		return -EINVAL;
+	}
```

Comment:
@plbossart, before bbf7d3b1c4f4 neither fe or be stream was checked, no error was generated if any of them was initialized at a later time.
Should we only protect the atomicity check and fixup with `if (fe_substream && be_substream) { }` ?

Reply:
If the BE substream can be initialized later, then the atomicity check is done in the wrong location. It would happen that we have a later initialization of an atomic BE and that would throw all kinds of errors.